Spec/RealtimeClientConnection.swift

```diff
@@ -2093,6 +2101,73 @@ class RealtimeClientConnection: QuickSpec {
                         expect(channel.errorReason).to(beIdenticalTo(protocolError.error))
                     }
 
+                    it("should resume the connection after an auth renewal and detach all channels if the connectionId changes") {
+                        let options = AblyTests.commonAppSetup()
+                        options.disconnectedRetryTimeout = 1.0
+                        options.tokenDetails = getTestTokenDetails(ttl: 2.0)
+                        let client = AblyTests.newRealtime(options)
+                        defer { client.dispose(); client.close() }
+
+                        let channel = client.channels.get("test")
+                        waitUntil(timeout: testTimeout) { done in
+                            channel.attach() { error in
+                                expect(error).to(beNil())
+                                done()
+                            }
+                        }
+
+                        let initialConnectionId = client.connection.id
+
+                        guard let firstTransport = client.transport as? TestProxyTransport else {
+                            fail("TestProxyTransport is not set"); return
+                        }
+
+                        // Wait for token to expire
+                        waitUntil(timeout: testTimeout) { done in
+                            let partialDone = AblyTests.splitDone(2, done: done)
+                            channel.once(.Detached) { error in
+                                expect(error).to(beNil())
+                                partialDone()
+                            }
+                            // Wait for connection resume
+                            client.connection.once(.Connected) { stateChange in
+                                expect(stateChange?.reason).to(beNil())
+                                partialDone()
+                            }
+                        }
+
+                        expect(firstTransport.protocolMessagesReceived.filter{ $0.action == .Disconnected }).to(haveCount(1))
+
+                        guard let secondsTransport = client.transport as? TestProxyTransport else {
+                            fail("TestProxyTransport is not set"); return
+                        }
+
+                        let connectedMessages = secondsTransport.protocolMessagesReceived.filter{ $0.action == .Connected }
+                        expect(connectedMessages).to(haveCount(1)) //New transport connected
+                        guard let receivedConnectionId = connectedMessages.first?.connectionId else {
+                            fail("ConnectionID is nil"); return
+                        }
+                        expect(client.connection.id).to(equal(receivedConnectionId))
+                        expect(client.connection.id).toNot(equal(initialConnectionId))
```

Comment:
@mattheworiordan the `connectionId` changes and that's why the channel gets detached.

First CONNECTED:
```
2016-11-23 13:16:04.508 xctest[32837:4092858] VERBOSE: RS:0x7f955c623000 ARTJsonLikeEncoder<msgpack>: protocolMessageFromDictionary {
    action = 4;
    connectionDetails =     {
        connectionKey = "e91RTOWlwA0vIq!tCwrawkZGCTS118T-ca6e91RTOWlwA0vIq";
        connectionStateTtl = 120000;
        maxFrameSize = 262144;
        maxIdleInterval = 15000;
        maxInboundRate = 250;
        maxMessageSize = 16384;
        serverId = "frontend.64d2.1.eu-central-1-A.i-d35f816e";
    };
    connectionId = tCwrawkZGC;
    connectionKey = "e91RTOWlwA0vIq!tCwrawkZGCTS118T-ca6e91RTOWlwA0vIq";
    connectionSerial = "-1";
}
```

It gets the DISCONNECTED from Ably:
```
2016-11-23 13:16:05.998 xctest[32837:4092858] VERBOSE: RS:0x7f955c623000 ARTJsonLikeEncoder<msgpack>: protocolMessageFromDictionary {
    action = 6;
    error =     {
        code = 40142;
        message = "Key/token status changed (expire)";
        statusCode = 401;
    };
}
```

The second CONNECTED:
```
2016-11-23 13:16:06.614 xctest[32837:4092858] VERBOSE: RS:0x7f955c623000 ARTJsonLikeEncoder<msgpack>: protocolMessageFromDictionary {
    action = 4;
    connectionDetails =     {
        connectionKey = "e91NTgaIgA0vIq!guMApg59VswaKAc4-d86e91NTgaIgA0vIq";
        connectionStateTtl = 120000;
        maxFrameSize = 262144;
        maxIdleInterval = 15000;
        maxInboundRate = 250;
        maxMessageSize = 16384;
        serverId = "frontend.25af.1.eu-central-1-A.i-305dca8c";
    };
    connectionId = guMApg59Vs;
    connectionKey = "e91NTgaIgA0vIq!guMApg59VswaKAc4-d86e91NTgaIgA0vIq";
    connectionSerial = "-1";
}
```

Full log: https://gist.github.com/ricardopereira/15a80610b22c9216d5587f2375359921